### PR TITLE
Add Https proxy system properties

### PR DESCRIPTION
**Pull request by davidh44**
## Motivation and Context
 Https Proxy system properties are currently not supported

Fixes https://github.com/aws/aws-sdk-java-v2/issues/1334

## Modifications

Added enum values for https proxy system settings in ProxySystemSetting. Updated ProxyConfiguration for apache, crt, and netty to check for https proxy properties

## Testing
 Added tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/aws-sdk-java-v2/blob/master/CONTRIBUTING.md) document
- [ ] Local run of `mvn install` succeeds
- [ ] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] I have added a changelog entry. Adding a new entry must be accomplished by running the `scripts/new-change` script and following the instructions. Commit the new file created by the script in `.changes/next-release` with your changes.
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: http-clients/apache-client/src/main/java/software/amazon/awssdk/http/apache/ProxyConfiguration.java [hunk 1]
```diff
@@ -96,7 +95,9 @@ public String scheme() {
      * @see Builder#password(String)
      */
     public String username() {
-        return resolveValue(username, ProxySystemSetting.PROXY_USERNAME);
+        String httpsPort = resolveValue(null, ProxySystemSetting.HTTPS_PROXY_HOST);
+        return httpsPort != null ? resolveValue(username, ProxySystemSetting.HTTPS_PROXY_USERNAME)
+               : resolveValue(username, ProxySystemSetting.PROXY_USERNAME);
```
Comment (zoewangg) [inline http-clients/apache-client/src/main/java/software/amazon/awssdk/http/apache/ProxyConfiguration.java:100]: Should we return usename based on the provided `scheme` instead of checking https first?

Same for other getters

File: utils/src/main/java/software/amazon/awssdk/utils/ProxySystemSetting.java [hunk 2]
```diff
@@ -27,7 +27,13 @@ public enum ProxySystemSetting implements SystemSetting  {
...
     NON_PROXY_HOSTS("http.nonProxyHosts"),
     PROXY_USERNAME("http.proxyUser"),
-    PROXY_PASSWORD("http.proxyPassword")
+    PROXY_PASSWORD("http.proxyPassword"),
+
+    HTTPS_PROXY_HOST("https.proxyHost"),
+    HTTPS_PROXY_PORT("https.proxyPort"),
+    HTTPS_NON_PROXY_HOSTS("https.nonProxyHosts"),
+    HTTPS_PROXY_USERNAME("https.proxyUser"),
+    HTTPS_PROXY_PASSWORD("https.proxyPassword")
```
Comment (zoewangg) [inline utils/src/main/java/software/amazon/awssdk/utils/ProxySystemSetting.java:36]: I don't think we need `HTTPS_NON_PROXY_HOSTS("https.nonProxyHosts")`


https://docs.oracle.com/javase/8/docs/technotes/guides/net/proxies.html

>As you probably guessed these work in the exact same manner as their http counterparts, so we won't go into much detail except to mention that the default port number, this time, is 443 and that for the "non proxy hosts" list, the HTTPS protocol handler will use the same as the http handler (i.e. http.nonProxyHosts).

**Review by zoewangg — COMMENTED** [review]
Is [AWS_Java_SDK_2_0.xml](https://github.com/aws/aws-sdk-java-v2/pull/3398/files#diff-55a9ae84fc35d634ce977aa8aa845a84a479905af2cdfbe9e9e396abf51d0241) commited accidentally?

File: utils/src/main/java/software/amazon/awssdk/utils/ProxySystemSetting.java (hunk 2 above)
Comment (davidh44) [reply utils/src/main/java/software/amazon/awssdk/utils/ProxySystemSetting.java]: Removed this enum

File: http-clients/apache-client/src/main/java/software/amazon/awssdk/http/apache/ProxyConfiguration.java (hunk 1 above)
Comment (davidh44) [reply http-clients/apache-client/src/main/java/software/amazon/awssdk/http/apache/ProxyConfiguration.java]: Updated to check scheme instead

File: http-clients/apache-client/src/main/java/software/amazon/awssdk/http/apache/ProxyConfiguration.java [hunk 3]
```diff
@@ -66,18 +67,17 @@ private ProxyConfiguration(DefaultClientProxyConfigurationBuilder builder) {
     }
 
     /**
-     * Returns the proxy host name either from the configured endpoint or
-     * from the "http.proxyHost" system property if {@link Builder#useSystemPropertyValues(Boolean)} is set to true.
+     * Returns the proxy host name either from the configured endpoint or from the "https.proxyHost" or "http.proxyHost" system
+     * property if {@link Builder#useSystemPropertyValues(Boolean)} is set to true.
```
Comment (zoewangg) [inline http-clients/apache-client/src/main/java/software/amazon/awssdk/http/apache/ProxyConfiguration.java:71]: Can we update Javadocs accordingly?

File: http-clients/apache-client/src/main/java/software/amazon/awssdk/http/apache/ProxyConfiguration.java [hunk 4]
```diff
@@ -184,8 +191,14 @@ public String toString() {
...
+            return endpoint.getHost();
+        }
+
+        if (Objects.equals(scheme(), "https")) {
+            return resolveValue(null, ProxySystemSetting.HTTPS_PROXY_HOST);
+        }
+        return resolveValue(null, ProxySystemSetting.PROXY_HOST);
     }
 
     private int resolvePort() {
```
Comment (zoewangg) [inline http-clients/apache-client/src/main/java/software/amazon/awssdk/http/apache/ProxyConfiguration.java:204]: It seems `resolvePort()` is invoked _before_ `resolveScheme` in the constructor, so scheme() is always null at this point.

We could move line 66 above line 64 and pass the scheme to `resolveHost(scheme)` and `resolvePort(scheme)`

File: .idea/inspectionProfiles/AWS_Java_SDK_2_0.xml [hunk 5]
```diff
@@ -180,6 +180,9 @@
     </inspection_tool>
     <inspection_tool class="JavaLangImport" enabled="true" level="WARNING" enabled_by_default="true" />
     <inspection_tool class="JavadocHtmlLint" enabled="true" level="ERROR" enabled_by_default="true" />
+    <inspection_tool class="JavadocReference" enabled="true" level="ERROR" enabled_by_default="true">
+      <option name="REPORT_INACCESSIBLE" value="false" />
+    </inspection_tool>
```
Comment (zoewangg) [inline .idea/inspectionProfiles/AWS_Java_SDK_2_0.xml:185]: Is this change intended?

File: http-clients/apache-client/src/main/java/software/amazon/awssdk/http/apache/ProxyConfiguration.java (hunk 4 above)
Comment (davidh44) [reply http-clients/apache-client/src/main/java/software/amazon/awssdk/http/apache/ProxyConfiguration.java:205]: Made changes accordingly

File: .idea/inspectionProfiles/AWS_Java_SDK_2_0.xml (hunk 5 above)
Comment (davidh44) [reply .idea/inspectionProfiles/AWS_Java_SDK_2_0.xml:185]: Reverted change

File: http-clients/apache-client/src/main/java/software/amazon/awssdk/http/apache/ProxyConfiguration.java (hunk 3 above)
Comment (davidh44) [reply http-clients/apache-client/src/main/java/software/amazon/awssdk/http/apache/ProxyConfiguration.java:72]: Updated Javadocs description

File: http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/ProxyConfiguration.java [hunk 6]
```diff
@@ -41,6 +41,7 @@ public final class ProxyConfiguration implements ToCopyableBuilder<ProxyConfigur
     private final String username;
     private final String password;
     private final Boolean useSystemPropertyValues;
+    private static final String HTTPS = "https";
```
Comment (zoewangg) [inline http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/ProxyConfiguration.java:44]: Can we use https://github.com/aws/aws-sdk-java-v2/blob/master/core/sdk-core/src/main/java/software/amazon/awssdk/core/Protocol.java#L29?  (Sorry, I should've mentioned it earlier)

File: .idea/inspectionProfiles/AWS_Java_SDK_2_0.xml [hunk 7]
```diff
@@ -238,7 +238,6 @@
     <inspection_tool class="NestedSwitchStatement" enabled="true" level="WARNING" enabled_by_default="true" />
     <inspection_tool class="NestedSynchronizedStatement" enabled="true" level="WARNING" enabled_by_default="true" />
     <inspection_tool class="NonExceptionNameEndsWithException" enabled="true" level="WARNING" enabled_by_default="true" />
-    <inspection_tool class="NonFinalFieldInEnum" enabled="true" level="WARNING" enabled_by_default="true" />
```
Comment (zoewangg) [inline .idea/inspectionProfiles/AWS_Java_SDK_2_0.xml:241]: Can we revert all changes in this file?

**Review by zoewangg — COMMENTED** [review]
Can we add a changelog entry?

File: http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/ProxyConfiguration.java (hunk 6 above)
Comment (davidh44) [reply http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/ProxyConfiguration.java:44]: core not accessible to http client modules

File: .idea/inspectionProfiles/AWS_Java_SDK_2_0.xml (hunk 7 above)
Comment (davidh44) [reply .idea/inspectionProfiles/AWS_Java_SDK_2_0.xml]: changes reverted

File: .changes/next-release/feature-AWSCommonRuntimeHTTPClient-dda0ac9.json [hunk 8]
```diff
@@ -0,0 +1,6 @@
+{
+    "category": "__AWS Common Runtime HTTP Client__",
```
Comment (zoewangg) [inline .changes/next-release/feature-AWSCommonRuntimeHTTPClient-dda0ac9.json:2]: nit, is there any reason `__` is used as prefix and suffix? This may not render correctly with our existing changelog formatting. Same as other changelog entries

File: .changes/next-release/feature-AWSCommonRuntimeHTTPClient-dda0ac9.json [hunk 9]
```diff
@@ -0,0 +1,6 @@
+{
+    "category": "__AWS Common Runtime HTTP Client__", 
+    "contributor": "", 
+    "type": "feature", 
+    "description": "Adds support for Https system properties"
```
Comment (zoewangg) [inline .changes/next-release/feature-AWSCommonRuntimeHTTPClient-dda0ac9.json:5]: Can we be more specific about what system properties we have added as part of this change? Same as other changelog entries

File: .changes/next-release/feature-AWSCommonRuntimeHTTPClient-dda0ac9.json (hunk 8 above)
Comment (davidh44) [reply .changes/next-release/feature-AWSCommonRuntimeHTTPClient-dda0ac9.json]: removed and updated

File: .changes/next-release/feature-AWSCommonRuntimeHTTPClient-dda0ac9.json (hunk 9 above)
Comment (davidh44) [reply .changes/next-release/feature-AWSCommonRuntimeHTTPClient-dda0ac9.json]: Updated to add descriptions

**Review by zoewangg — APPROVED** [review]
